File: Data Structures & Algorithms/palindromic-substrings/submission-1.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        res  = 0
        for i in range(len(s)):
            p1,p2 = i,i
            while p1>=0 and p2<len(s) and s[p1] == s[p2]:
                res+=1
                p1 -= 1
                p2 += 1
            if i<len(s)-1:
                p1,p2 = i,i+1
                while  p1>=0 and p2<len(s) and s[p1] == s[p2]:
                    res+=1
                    p1 -= 1
                    p2 += 1
        return res

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<string,vector<string>> m;
        for (auto s: strs){
            string t = s;
            std::sort(t.begin(),t.end());
            m[t].push_back(s);
        }
        std::vector<std::vector<string>> res;
        for (const auto p:m){
            res.push_back(p.second);
        }
        return res;
    }
};
